include/raft.h

```diff
@@ -104,6 +107,7 @@ struct raft_server
 
 /**
  * Hold information about all servers currently part of the cluster.
+ * WARNING: This struct is encoded/decoded, be careful when adapting it.
  */
 struct raft_configuration
```

Comment:
I think it could make sens to hide this one, there are already `raft_configuration_init` and `raft_configuration_close` anyway. `dqlite` likes to access the configuration members directly currently so we would need to add a bit of API to facilitate that.

Reply:
I think `struct raft_configuration` needs to be public, because it's part of `struct raft_snapshot` which any `raft_io` implementation (potentially third party) must be able to serialize.